packages/kit/src/exports/vite/index.js

```diff
@@ -341,6 +341,28 @@ async function kit({ svelte_config }) {
 					]
 				}
 			};
+			if (is_rolldown && new_config.optimizeDeps) {
+				delete new_config.optimizeDeps.esbuildOptions; // vite 8 logs a warning when esbuildOptions is used
```

Comment:
rather than deleting it, we should probably only add it if `!is_rolldown`

Reply:
refactored to nested conditionals for adding the plugins only when they are needed